(maint) Call `exist?` instead of `exists?`

mhashizume opened the pull request:
The `exists?` method has long been deprecated and has been officially removed as of Ruby 3.2.0.

ekohl reviewed (commented):
I think some stubbing/mocking fails.

mhashizume commented:
Thanks for the feedback @ekohl . I've addressed the spec issues and added another commit to address a minor issue I also ran into while working on this.
